app/src/main/java/com/example/sharingang/NewItemFragment.kt

```diff
@@ -1,43 +1,95 @@
+    private var cancellationTokenSource = CancellationTokenSource()
+    private val requestPermissionLauncher = requestPermissionLauncher(
+        this
+    ) {
+        doOrGetPermission(
+            this.context,
+            this,
+            Manifest.permission.ACCESS_FINE_LOCATION,
+            { updateLocation() }, null
+        )
+    }
+
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
 
-        val binding: FragmentNewItemBinding =
+        binding =
             DataBindingUtil.inflate(inflater, R.layout.fragment_new_item, container, false)
-
         binding.createItemButton.setOnClickListener { view: View ->
             viewModel.addItem(
                 Item(
-                        price = binding.price?.toDoubleOrNull() ?: 0.0,
-                        description = binding.description ?: "",
-                        title = binding.title ?: "",
-                        category = binding.categorySpinner.selectedItemPosition,
-                        categoryString = resources.getStringArray(R.array.categories)[binding.categorySpinner.selectedItemPosition]
+                    price = binding.price?.toDoubleOrNull() ?: 0.0,
+                    description = binding.description ?: "",
+                    title = binding.title ?: "",
+                    category = binding.categorySpinner.selectedItemPosition,
+                    categoryString = resources.getStringArray(R.array.categories)[binding.categorySpinner.selectedItemPosition],
+                    latitude = binding.latitude?.toDoubleOrNull() ?: 0.0,
+                    longitude = binding.longitude?.toDoubleOrNull() ?: 0.0
                 )
             )
             view.findNavController().navigate(R.id.action_newItemFragment_to_itemsListFragment)
         }
+
+        setupLocation()
         return binding.root
     }
+
+    private fun setupLocation(){
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireContext())
+        binding.newItemGetLocation.setOnClickListener {
+            doOrGetPermission(
+                this.context,
+                this,
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                { updateLocation() },
+                requestPermissionLauncher
+            )
+        }
+    }
+    @SuppressLint("MissingPermission")
+    private fun updateLocation() {
+        val getLocationTask = fusedLocationClient.getCurrentLocation(
+            LocationRequest.PRIORITY_HIGH_ACCURACY,
+            cancellationTokenSource.token
+        )
+        getLocationTask.addOnCompleteListener {
+            if (it.isSuccessful && it.result != null) {
+                binding.latitude = it.result.latitude.toString()
+                binding.longitude = it.result.longitude.toString()
+            }
+        }
+    }

```

Comment:
I think that maybe this could be extracted further, given the duplication with the other fragment?

Reply:
You're right, I will see what I can do with this